src/WebJobs.Script.WebHost/WebJobsScriptHostService.cs

```diff
@@ -1047,6 +1067,99 @@ private void EndStartupOperation(ScriptHostStartupOperation operation)
             _logger.StartupOperationCompleted(operation.Id);
         }
 
+        private async Task LogErrorWithTransientOtelLogger(string log)
+        {
+            var loggerFactory = BuildOtelLoggerFactory();
+
+            if (loggerFactory is not null)
+            {
+                var logger = loggerFactory.CreateLogger(ScriptConstants.LogCategoryHostGeneral);
+                logger.LogError(log);
+
+                // Delay increases the chance that the log is sent to AppInsights/Otel before the logger factory is disposed
+                await Task.Delay(2000);
+                // Do a force flush as the host is shutting down and we want to ensure the logs are sent before disposing the logger factory
+                ForceFlush(loggerFactory);
+                // Give some time for the logger to flush
+                await Task.Delay(4000);
+            }
+        }
+
+        private ILoggerFactory BuildOtelLoggerFactory()
+        {
+            var appInsightsConnStr = GetConfigurationValue(AppInsightsConnectionString, _config);
+            var otlpEndpoint = GetConfigurationValue(OtlpEndpoint, _config);
+
+            if (appInsightsConnStr is not { Length: > 0 } && otlpEndpoint is not { Length: > 0 })
```

Comment:
What happens when AppInsights connection string is configured, but Otel endpoint is not. Should we return from this method in that case, or continue to use AppInsights connection string to log the error?

Reply:
Users will have either an AppInsights endpoint, an OTLP endpoint, or both. Based on the configured settings, logs will be forwarded accordingly.